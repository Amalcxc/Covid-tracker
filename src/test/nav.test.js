import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Nav from '../components/nav';

describe('nav component', () => {
  test('link element', () => {
    render(
      <Router>
        <Nav />
      </Router>,
    );
    return expect(screen.getByRole('link')).toBeInTheDocument;
  });
});

describe('test with jest snapshot', () => {
  it('render correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Nav />
        </Router>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});