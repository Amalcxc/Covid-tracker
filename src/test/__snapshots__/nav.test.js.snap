// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with jest snapshot render correctly 1`] = `
<div>
  <nav
    className="nav row isBlue"
  >
    <div
      className="col-4 pt-3"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <h1
          className="p-1 c-white"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 8a.5.5 0 0 0-.5-.5H2.707l3.147-3.146a.5.5 0 1 0-.708-.708l-4 4a.5.5 0 0 0 0 .708l4 4a.5.5 0 0 0 .708-.708L2.707 8.5H14.5A.5.5 0 0 0 15 8z"
              fillRule="evenodd"
            />
          </svg>
        </h1>
      </a>
    </div>
    <div
      className="icons c-white"
    >
      <h4>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
          />
          <path
            d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
          />
        </svg>
      </h4>
    </div>
  </nav>
</div>
`;
